### PR TITLE
PR: Fix updates with complex values on variable explorer

### LINKED ISSUE #7848: Variable editor problems on clicking
### What steps will reproduce the problem?

When I double click on a complex variable in a 40x40 complex double array in the variable editor, this problem occured.

### Traceback
```python-traceback
  File "C:\ProgramData\Anaconda3\lib\site-packages\spyder\widgets\variableexplorer\arrayeditor.py", line 331, in setData
    if val < self.vmin:
TypeError: '<' not supported between instances of 'float' and 'complex'
WARNING:spyder.widgets.github.backend:failed to send bug report on github. response={'code': 401, 'json': {'message': 'Bad credentials', 'documentation_url': 'https://developer.github.com/v3'}}
```

## Versions

* Spyder version: 3.3.0 
* Python version: 3.6.5
* Qt version: 5.9.4
* PyQt5 version: 5.9.2
* Operating System: Windows 10

### Dependencies

```
pyflakes >=0.6.0  :  1.6.0 (OK)
pycodestyle >=2.3 :  2.4.0 (OK)
pygments >=2.0    :  2.2.0 (OK)
sphinx >=0.6.6    :  1.7.2 (OK)
rope >=0.9.4      :  0.10.7 (OK)
jedi >=0.9.0      :  0.12.0 (OK)
nbconvert >=4.0   :  5.3.1 (OK)
pandas >=0.13.1   :  0.22.0 (OK)
numpy >=1.7       :  1.14.2 (OK)
sympy >=0.7.3     :  1.1.1 (OK)
cython >=0.21     :  0.28.2 (OK)
qtconsole >=4.2.0 :  4.3.1 (OK)
IPython >=4.0     :  6.3.1 (OK)
matplotlib >=2.0.0:  2.2.2 (OK)
pylint >=0.25     :  1.8.4 (OK)
```

### PR DESCRIPTION
**Pull request by goanpeca**
## Description of Changes

* [x] Wrote at least one-line docstrings (for any new functions)
* [x] Added unit test(s) covering the changes (if testable)

### Issue(s) Resolved

Fixes #7848

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @goanpeca

### CODE REVIEW

File: spyder/plugins/variableexplorer/widgets/arrayeditor.py [hunk 1]
```diff
@@ -850,10 +855,13 @@ def current_dim_changed(self, index):
...
-        """Reimplement Qt method"""
-        for index in range(self.stack.count()):
-            self.stack.widget(index).accept_changes()
-        QDialog.accept(self)
+        """Reimplement Qt method."""
+        try:
+            for index in range(self.stack.count()):
+                self.stack.widget(index).accept_changes()
+            QDialog.accept(self)
+        except RuntimeError:
```
Comment (dalthviz) [inline spyder/plugins/variableexplorer/widgets/arrayeditor.py]: @goanpeca maybe a docstring with a description of the need of this except could be added?

Comment (goanpeca) [reply spyder/plugins/variableexplorer/widgets/arrayeditor.py]: Will do! thanks :-)

**Comment by ccordoba12** [comment]
@goanpeca, could you finish this PR so we can include it in 4.0.1? Thanks!

**Comment by goanpeca** [comment]
Will do later today!

File: spyder/plugins/variableexplorer/widgets/arrayeditor.py (hunk 1 above)
Comment (goanpeca) [reply spyder/plugins/variableexplorer/widgets/arrayeditor.py]: Fixed

**Comment by goanpeca** [comment]
@ccordoba12 addressed @dalthviz comment. This is ready.

**Pull request merged**